nautobot/core/middleware.py

```diff
@@ -24,12 +32,37 @@ def header(self):
     def process_request(self, request):
 
         # Bypass middleware if remote authentication is not enabled
-        if not settings.REMOTE_AUTH_ENABLED:
+        if "nautobot.core.authentication.RemoteUserBackend" not in settings.EXTERNAL_AUTH_BACKENDS:
             return
 
         return super().process_request(request)
 
 
+class ExternalAuthMiddleware(AuthenticationMiddleware):
+    """
+    Custom implementation of Django's AuthenticationMiddleware used to set permissions for ephemeral users
+    """
+
+    def process_request(self, request):
+
+        # Bypass middleware if remote authentication is not enabled
+        if not settings.EXTERNAL_AUTH_BACKENDS:
+            return
+        # side effect magic? ¯\_(ツ)_/¯
+        logger = logging.getLogger("nautobot.authentication.RemoteUserBackend")  # noqa: F841
```

Comment:
@jathanism I have no idea what this line does.  
https://github.com/nautobot/nautobot/blob/develop/nautobot/core/authentication.py#L106
I added it here, but I'm not sure if it is still relevant.

Reply:
Yeah that seems like a strange place for a logger to be defined. It looks like a side effect of the underlying functions called inside of `configure_user` having been moved. If either of those functions were to be called out of this scope, they'd be using the module-global logger and not this one. IMO just remove it.